src/main/java/seedu/address/model/student/Student.java

```diff
@@ -0,0 +1,134 @@
+package seedu.address.model.student;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Student in the record.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Student {
+
+    // Student fields
+    private final Name studentName;
+    private final Id id;
+
+    // Parent fields
+    private final Name parentName;
+    private final Phone phone;
+    private final Address address;
+
+    // Additional fields - may not be implemented
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Student(Name studentName, Id id, Name parentName, Phone phone, Address address, Set<Tag> tags) {
+        requireAllNonNull(studentName, id, parentName, phone, address, tags);
+        this.studentName = studentName;
+        this.id = id;
+        this.parentName = parentName;
+        this.phone = phone;
+        this.address = address;
+        this.tags.addAll(tags);
+    }
+
+    public Name getStudentName() {
+        return studentName;
+    }
+
+    public Id getId() {
+        return id;
+    }
+
+    public Name getParentName() {
+        return parentName;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public Address getAddress() {
+        return address;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both students have the same name.
+     * This defines a weaker notion of equality between two students.
+     */
+    public boolean isSameStudent(Student otherStudent) {
```

Comment:
Would the method name "hasSameName" be better here? The method name "isSameStudent" seems like equality.

Reply:
Good point! Will look into it in the next PR 👍🏻

Reply:
I will update this method name to be hasSameNameOrId in my next PR so that if students have the same name or ID, they are considered equal, and cannot be added.

Reply:
Yes that makes sense! Will need to add in a few more test cases for the modified method.